genie-agent/src/main/java/com/netflix/genie/agent/jobexecution/statemachine/stateactions/CleanupJobAction.java

```diff
@@ -0,0 +1,46 @@
+/*
+ *
+ *  Copyright 2018 Netflix, Inc.
+ *
+ *     Licensed under the Apache License, Version 2.0 (the "License");
+ *     you may not use this file except in compliance with the License.
+ *     You may obtain a copy of the License at
+ *
+ *         http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *     Unless required by applicable law or agreed to in writing, software
+ *     distributed under the License is distributed on an "AS IS" BASIS,
+ *     WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *     See the License for the specific language governing permissions and
+ *     limitations under the License.
+ *
+ */
+
+package com.netflix.genie.agent.jobexecution.statemachine.stateactions;
+
+import com.netflix.genie.agent.jobexecution.statemachine.Events;
+import com.netflix.genie.agent.jobexecution.statemachine.States;
+import lombok.extern.slf4j.Slf4j;
+import org.springframework.context.annotation.Lazy;
+import org.springframework.statemachine.StateContext;
+import org.springframework.stereotype.Component;
+
+/**
+ * Action performed when in state CLEANUP_JOB.
+ * @author mprimi
+ * @since 4.0.0
+ */
+@Slf4j
+@Component
+@Lazy
+class CleanupJobAction extends BaseStateAction implements StateAction.CleanupJob {
+
+    /**
+     * {@inheritDoc}
+     */
+    @Override
+    protected Events executeStateAction(final StateContext<States, Events> context) {
+        log.info("Cleaning up job...");
```

Comment:
You might want to add `TODO` in all of these just so we can quickly track which need to be done

Reply:
Will do